File: test/translations/classes/constructor_parameter_shadows_field.java
public class MyResource {
    public final String env;
    private final String resource;

    public MyResource(String resource) {
        this.resource = resource;
        this.env = "production";
    }

    public String description() {
        return this.resource;
    }
}

MyResource r = new MyResource("bucket");

File: test/translations/classes/constructor_parameter_shadows_field.go
type myResource struct {
	env *string
	resource *string
}

func newMyResource(resource *string) *myResource {
	this := &myResource{}
	this.resource = resource
	this.env = jsii.String("production")
	return this
}

func (this *myResource) description() *string {
	return this.resource
}

r := newMyResource(jsii.String("bucket"))
